anndata/core/anndata.py

```diff
@@ -39,7 +40,8 @@
     asview,
     _resolve_idxs,
 )
-from .. import h5py, utils
+from .sparsedataset import SparseDataset
```

Comment:
`sparse_dataset`, only stdlib names are `nounderscore`

Reply:
> only stdlib names are nounderscore

I'm not sure that's a common convention. [PEP-8 recommends against underscores in module names](https://www.python.org/dev/peps/pep-0008/#id40).

> Or maybe hdf5_sparse

I'm pretty sure you can currently just pass a zarr group to SparseDataset and it'd work. This works:

```python
import anndata as ad
import scanpy as sc
import zarr

pbmc = sc.datasets.pbmc3k()
pbmc.write_zarr("pbmc.zarr")
zx = ad.core.sparsedataset.SparseDataset(zarr.open("pbmc.zarr/X"))
assert not np.any((zx[50:100, :] != pbmc[50:100, :].X).toarray())
```

> no reason the module has to be called the same as the class

I am wondering why we have separate classes for `SparseDataset` and `backed_csc_matrix`/ `backed_csr_matrix`. We could just have `CSRDataset` and `CSCDataset`.

Reply:
> PEP-8 recommends against underscores in module names.

Yes, and PEP 8 is a style guide … for the stdlib, not 3rd party modules. Common misconception.

> I'm pretty sure you can currently just pass a zarr group to SparseDataset and it'd work.

Right, zarr isn’t HDF5-based. Is there a nice concise word for what they both are? They’re both supposed to be an efficient memory layout for binary data, with focus on performance and compressability, right? But that’d make a shitty module name :wink:

Reply:
> Yes, and PEP 8 is a style guide … for the stdlib, not 3rd party modules.

I mean, it's a style guide which the standard library uses, but so do many packages. Just going off convention from here:

* `readwrite` not `read_write`
* `h5sparse` not `h5_sparse`
* `scanpydoc` not `scanpy_doc`

I also prefer `from sparsedataset import SparseDataset` to `from sparse_dataset import SparseDataset`.

Ultimately, it's an internal module so I guess it doesn't matter much.

> Is there a nice concise word for what they both are? 

Hierarchical array store?

> Or maybe (don’t know how feasible), remove mentions of rows and columns there and make other code use SparseDataset in terms of major_axis and minor_axis.

I think I might just wait and watch for what pydata/sparse does here. I do like having separate classes for dispatch – since we want to do different things depending on layout. If we could dispatch on generic types, that might be different.

Also I'd want a usability/ performance reason to do any major overhauls of this code.

Reply:
Let’s call it it “storage” then!

Reply:
I'd rather keep the name specific to the data structure until it's diverged enough we can just get rid of the old licensing.